pkg/volume/gce_pd/gce_pd.go

```diff
@@ -109,7 +124,31 @@ func (plugin *gcePersistentDiskPlugin) GetVolumeLimits() (map[string]int64, erro
 	}
 
 	volumeLimits := map[string]int64{
-		gceVolumeLimitKey: 16,
+		gceVolumeLimitKey: DiskAttachLimit[Default],
+	}
+	instances, ok := cloud.Instances()
+	if !ok {
+		glog.Warning("Failed to get instances from cloud provider")
+		return volumeLimits, nil
+	}
+
+	instanceType, err := instances.InstanceType(context.TODO(), plugin.host.GetNodeName())
+	if err != nil {
+		glog.Errorf("Failed to get instance type from GCE cloud provider")
+		return volumeLimits, nil
+	}
+	if strings.HasPrefix(instanceType, "n1-standard-") || strings.HasPrefix(instanceType, "n1-highmem-") || strings.HasPrefix(instanceType, "n1-highcpu-") {
```

Comment:
There are also "n1-ultramem-" machine types now.  Would it be better to put all the machine type prefixes into a set?

Will this work with custom machine types? Maybe for beta it would be better to query the actual cpus available on the node instead of deriving it from the machine type name.

Reply:
instances with custom machine types are still limited to 16 disk volumes per instance.